### PR TITLE
added tls_config for downstream query

### LINKED ISSUE #6674: [query-frontend] TLS configarion for downstream query backend
### Is your proposal related to a problem?

I'm setting up Thanos in an environment where Istio enforces strict mTLS. Each pod, including thanos frontend, runs a sidecar that exposes the application through a TLS enabled port with self signed certificates generated by Istio.

Most Thanos and Prometheus components lets me configure the application to use ca, cert and key from a file on disk. This is very conventient because the Istio sidecar can provide a volume with these files.

But I can't find a way to pass this configuration to query-frontend. It seems like the binary simply does not take any argument for this. We have `--query-frontend.downstream-tripper-config` and `--query-frontend.downstream-tripper-config-file` but these only provide a subset of the config that the transport go library can handle. 

### Describe the solution you'd like

I would like to pass a TLSConfig to Thanos query-frontend pointing out the caFile, certFile and keyFile on disk as well as the ability to skip hostname check. These could be provided through `--query-frontend.downstream-tripper-config` and `--query-frontend.downstream-tripper-config-file`

For Ruler I can accomplish this with a query_config:
```
  - http_config:
      tls_config:
        ca_file: "/etc/certs/root-cert.pem"
        cert_file: "/etc/certs/cert-chain.pem"
        key_file: "/etc/certs/key.pem"
        insecure_skip_verify: true
```

### Describe alternatives you've considered

The only way around this that I have found is to add pod annotations so that istio ignores traffic on the incoming http port of thanos query backend and on the outgoing port on query frontend. But this affects other services such as Grafana who also communcates with query backend. Grafana can handle these tls settings but I have to make the same hacky workaround on the Grafana pod anyway since I've had to bypass it on the datasource.

### PR DESCRIPTION
**Pull request by Player256**
Fixes #6674

### CODE REVIEW

File: pkg/queryfrontend/config.go [hunk 1]
```diff
@@ -186,6 +187,7 @@ type DownstreamTripperConfig struct {
 	MaxIdleConns          *int               `yaml:"max_idle_conns"`
 	MaxIdleConnsPerHost   *int               `yaml:"max_idle_conns_per_host"`
 	MaxConnsPerHost       *int               `yaml:"max_conns_per_host"`
+	TLSConfig             *tls.Config        `yaml:"tls_config"`
```
Comment (saswatamcode) [inline pkg/queryfrontend/config.go:190]: I don't think this is the correct struct for TLS configs. Should be `exthttp.TLSConfig`, which you can then process and pass to `downstreamTripper.TLSClientConfig` by using `exthttp.NewTLSConfig()`

**Review by saswatamcode — COMMENTED** [review]
Thanks!

Small comment around config struct to use. Also, could you sign the commit?

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,8 +16,12 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 
 - [#6605](https://github.com/thanos-io/thanos/pull/6605) Query Frontend: Support vertical sharding binary expression with metric name when no matching labels specified.
 - [#6308](https://github.com/thanos-io/thanos/pull/6308) Ruler: Support configuration flag that allows customizing template for alert message.
+<<<<<<< HEAD
```
Comment (Kickball) [inline CHANGELOG.md:19]: It looks like some Git artefacts made their way into the commit.

**Comment by saswatamcode** [comment]
Let's close this one in favour of #6760 @Player256

**Pull request closed without merging**